apps/sync/js/crypto/stringconversion.js

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* exported
+  StringConversion
+*/
+
+var StringConversion = (() => {
+  return {
+    rawStringToByteArray(str) {
+      if (typeof str != 'string') {
+        throw new Error('Not a string');
+      }
+      var strLen = str.length;
+      var byteArray = new Uint8Array(strLen);
+      for (var i = 0; i < strLen; i++) {
+        byteArray[i] = str.charCodeAt(i);
+      }
+      return byteArray;
+    },
+
+    base64StringToByteArray(base64) {
+      if (typeof base64 != 'string' || base64.length % 4 !== 0) {
+        throw new Error(`Number of base64 digits must be a multiple of 4 to con\
+vert to bytes`);
+      }
+      return StringConversion.rawStringToByteArray(window.atob(base64));
+    },
+
+    hexStringToByteArray(hexStr) {
+      if (typeof hexStr != 'string' || hexStr.length % 2 !== 0) {
+        throw new Error(`Must have an even number of hex digits to convert to b\
+ytes`);
+      }
+      var numBytes = hexStr.length / 2;
+      var byteArray = new Uint8Array(numBytes);
+      //FIXME: Can this be done faster?
```

Comment:
I had a look around to see if I could find alternative algorithms for this, but this seems to be the only way. I made [a note on the Sync app integration bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1200539#c2) to remind myself that performance of the Sync app as a whole needs testing, to identify if and where it's slow.